lib/std/os.zig

```diff
@@ -5215,6 +5222,49 @@ pub fn getFdPath(fd: fd_t, out_buffer: *[MAX_PATH_BYTES]u8) RealPathError![]u8 {
             };
             return target;
         },
+        .freebsd => {
+            const allocator = std.heap.c_allocator;
```

Comment:
Unless I'm misunderstanding something:
- `c_allocator` is only available when linking libc
- Hard coding an allocator within a function like this goes against Zig's conventions on how allocators are used (it should be provided as a parameter instead)

Reply:
Unless I am misunderstanding something (highly likely), calling sysctl implies linking libc.

If that is not the case, adding an allocator parameter would break compatibility with all existing code that calls this function.  Since it is a public function, I would think that would be undesirable.

At the moment, only the FreeBSD specific code needs an allocator.  I do not want to put in code that does not adhere to Zig's conventions.  Is it better to hard code an allocator and not break compatibility or is it better to add an allocator parameter?  Or, is there another option?

Reply:
If linking libc is implied then ignore that bullet point.

> Is it better to hard code an allocator and not break compatibility

AFAIK there's nowhere in the Zig standard library that does this currently. Going from a non-allocating API to an allocating one is always a breaking change, and for cross-platform functions it "infects" the cross-platform API (see things like `std.process.args` and `std.process.argsWIthAllocator` where `argsWithAllocator` is the cross-platform API even though the allocator is only needed on Windows and WASI).

Reply:
Actually, I have found a function in the standard library that does hard-code the `c_allocator`, so I might be wrong:

https://github.com/ziglang/zig/blob/1653a9b2597c66cbcc88ea75d8a4b88c163584a5/lib/std/Thread.zig#L654-L656

(also worth noting that it's probably good to take my input with a grain of salt; it might be worth waiting for someone with more knowledge about this particular implementation to weigh in)

Reply:
I believe that that particular line in Thread.zig is what led me to hard code the allocator in my code.  It does appear that it is the only place where that happens.  I can see arguments for not hard coding the allocator and it is something I would think people should avoid.

I'll see if any more people weigh in on this.  Having 2 flavors of a function, 1 with and 1 without an allocator, seems clunky to me but I am willing to follow wiser counsel on this.